### PR TITLE
vpa max

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/giantswarm/issues/14999

This PR computes the max memory allowed for a Prometheus pod.

max memory = lowest node allocatable memory * 0.9

```console
$ k get no -lrole=worker -oyaml|yq -r '.items[].status.allocatable.memory'
15917740Ki
15745708Ki
15745708Ki
15917740Ki
```

`15745708 * 0.9 = 14171137.2`

```console
$ k get vpa -n 8jhhc-prometheus prometheus -oyaml | yq -y -r '.spec.resourcePolicy.containerPolicies[].maxAllowed'
memory: 14325966Ki
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -8,7 +8,7 @@ require (
 	github.com/Masterminds/sprig v2.22.0+incompatible
 	github.com/giantswarm/apiextensions/v2 v2.6.2
 	github.com/giantswarm/backoff v0.2.0
-	github.com/giantswarm/k8sclient/v4 v4.0.0
+	github.com/giantswarm/k8sclient/v4 v4.0.1-0.20210112105216-2a9eb31d062a
```
Comment (TheoBrigitte) [inline go.mod:11]: reminder: bump this to v4.1.0

**Comment by QuentinBisson** [comment]
Is your initial description still valid as it seems that the max memory in the spec is the highest and not multiplied by 0.9 even though the PR says the opposite?

**Comment by TheoBrigitte** [comment]
> Is your initial description still valid as it seems that the max memory in the spec is the highest and not multiplied by 0.9 even though the PR says the opposite?

Just re-tested and changed the description.

**Review by QuentinBisson — APPROVED** [review]

**Comment by QuentinBisson** [comment]
It works for me. We can always adjust the coefficient if daemon set gets bigger

**Comment by TheoBrigitte** [comment]
This was a painful PR, happy to close it :)

**Comment by JosephSalisbury** [comment]
this is vvv nice

**Pull request merged**